beacon-chain/sync/utils.go

```diff
@@ -49,6 +49,19 @@ func (s *Service) dedupBlocksAndRoots(blks []*ethpb.SignedBeaconBlock, roots [][
 	return newBlks, newRoots, nil
 }
 
+func (s *Service) dedupRoots(roots [][32]byte) [][32]byte {
```

Comment:
Is this tested? I think we can add a duplicated root in `TestService_BatchRootRequest` to cover this function?

Reply:
great idea, will add that in